main.go

```diff
@@ -89,8 +90,17 @@ func main() {
 		log.SetOutput(f)
 	}
 
+	sigKillChan := make(chan os.Signal, 1)
+	signal.Notify(sigKillChan, os.Interrupt, os.Kill)
+
 	var wg sync.WaitGroup
 	proxyListener := NewProxyListener(config, &wg, watchMode)
 	proxyListener.StartListeners()
-	wg.Wait()
+	defer proxyListener.StopListeners()
+	for {
+		select {
+		case <-sigKillChan:
```

Comment:
right now we only listen for sig kill but i could imagine in the future a HUP signal could cause the configuration to be reloaded...